bin/deploy.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+ENV=$1
+RELEASE_TAG=$(git rev-parse HEAD)
+APP_NAME=react-london
+AWS_ACCOUNT=578418881509
+AWS_REGION=eu-west-1
+ECR_REPO=$AWS_ACCOUNT.dkr.ecr.$AWS_REGION.amazonaws.com/$APP_NAME
+EB_BUCKET=elasticbeanstalk-$AWS_REGION-$AWS_ACCOUNT
+
+if [ -z "$ENV" ] || [ -z "$RELEASE_TAG" ]
+then
+  echo Usage:
+  echo "  deploy.sh ENVIRONMENT"
+  exit 1
+fi
+
+set -eu
+set -o pipefail
+
+# Authenticate
+eval $(aws ecr get-login --region=$AWS_REGION)
+
+# Build Docker image
+if [ "$ENV" == "production" ]
+then
+  VERSION=$RELEASE_TAG
+  docker build -t $APP_NAME:$RELEASE_TAG .
+  docker tag $APP_NAME:$RELEASE_TAG $ECR_REPO:$RELEASE_TAG
+  docker push $ECR_REPO:$RELEASE_TAG
+else
+  VERSION=$ENV
+  docker build -t $APP_NAME:$ENV .
+  docker tag $APP_NAME:$ENV $ECR_REPO:$ENV
+  docker push $ECR_REPO:$ENV
+fi
+
+# Apply docker image path to Dockerrun.aws.json template
+cp Dockerrun.aws.json.template Dockerrun.aws.json
+perl -pi -e "s,<ECR_REPO>,$ECR_REPO,g" "Dockerrun.aws.json"
+perl -pi -e "s,<TAG>,$VERSION,g" "Dockerrun.aws.json"
+
+# Zip up Dockerrun.aws.json and upload to S3 bucket
+ZIP_FILE=$VERSION.zip
+zip -r $ZIP_FILE Dockerrun.aws.json
+aws s3 cp $ZIP_FILE s3://$EB_BUCKET/$APP_NAME/$ZIP_FILE
+
+# In production, create a new app version
+if [ "$ENV" == "production" ]
+then
+  # Create a new application version with the zipped up Dockerrun file
+  aws elasticbeanstalk create-application-version \
+    --application-name $APP_NAME \
+    --version-label $VERSION \
+    --source-bundle S3Bucket=$EB_BUCKET,S3Key=$APP_NAME/$ZIP_FILE \
```

Comment:
You probably know this already but just to note that in S3 this means your deployed zip file will live in a sub folder, e.g. `react.london/<filename>.zip`. Can't remember if you have to create the subfolder prior to deploying or if it creates it on the fly.